elastic/beats
# Add max bytes in line limit

## What does this PR do?

Handles OOM on long lines gracefully, writes the error into log.

- Addresses https://github.com/elastic/beats/issues/19500

## Why is it important?

Handle bad data gracefully without OOM.

## Checklist


## Author's Checklist

- [ ] I just recently started looking at the beats code so might not have enough knowledge of any possible side-effects of this change. Please review with that in mind.

## How to test this PR locally

Refer to https://github.com/elastic/beats/issues/19500 for test case.
The filebeat doesn't get killed due to OOM with this change and the error is logged (see the screenshot attached).

## Related issues

- Addresses https://github.com/elastic/beats/issues/19500

## Screenshots

<img width="1110" alt="Screen Shot 2020-07-01 at 9 16 06 AM" src="https://user-images.githubusercontent.com/872351/86249080-c5f92600-bb7c-11ea-9703-91d06e397fa8.png">

## Logs

```
2020-07-01T13:15:17.630Z	INFO	[crawler]	beater/crawler.go:108	Loading and starting Inputs completed. Enabled inputs: 1
2020-07-01T13:15:17.633Z	INFO	log/harvester.go:297	Harvester started for file: /test/readme.log
2020-07-01T13:15:17.635Z	ERROR	log/harvester.go:330	Read line error: exceeded max bytes in line limit; File: /test/readme.log
```
